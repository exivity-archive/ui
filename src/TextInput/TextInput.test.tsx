import * as React from 'react'
import { mountWithTheme } from '../utils/tests/mountWithTheme'

import { TextInput } from '.'

test('value gets changed after an on change event', () => {
  const initialValue = 'initialValue'
  const updatedValue = 'changed'

  const onChangeMock = jest.fn(x => x)

  const textInput = mountWithTheme(<TextInput value={initialValue} onChange={onChangeMock}/>)

  textInput
    .find('input')
    .simulate('change', { target: { value: updatedValue } })

  expect(onChangeMock.mock.results[0].value).toBe(updatedValue)
})
