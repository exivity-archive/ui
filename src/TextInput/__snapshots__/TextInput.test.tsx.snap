// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput snapshot 1`] = `
.c0 {
  font-family: 'Fira Sans',sans-serif;
  font-weight: normal;
  font-size: 16px;
  color: #222;
  line-height: 1.5;
  font-size: 16px;
  display: block;
  box-sizing: border-box;
  width: 100%;
  padding: 0 0.5em;
  height: 2.5em;
  border-radius: 4px;
  outline: 0;
  border: 0;
  border: 1px solid #eee;
  background-color: #eee;
  --focus-color: 153,153,153;
}

.c0:hover {
  border-bottom: 1px solid rgba(var(--focus-color),0.5);
}

.c0:focus {
  border-bottom: 1px solid rgba(var(--focus-color),1);
}

.c0::-webkit-input-placeholder {
  color: currentcolor;
  opacity: 0.5;
}

.c0::-moz-placeholder {
  color: currentcolor;
  opacity: 0.5;
}

.c0:-ms-input-placeholder {
  color: currentcolor;
  opacity: 0.5;
}

.c0::placeholder {
  color: currentcolor;
  opacity: 0.5;
}

.c0[disabled] {
  cursor: not-allowed;
  box-shadow: inset 0 0 999em rgba(128,128,128,0.2);
}

<input
  className="c0"
  onChange={[Function]}
  secondary={true}
  type="text"
  value=""
/>
`;
